src/middlewares/responseSchema.middleware.ts

```diff
@@ -0,0 +1,46 @@
+import { NextFunction, Request, Response } from 'express';
+import { dayjs } from '@utils/dayjs.util';
+import { serializePagination } from '@utils/serializePagination.util';
+
+export const responseSchema = () => {
+  return (body: Object, req: Request, res: Response) => {
+    const timestamp = dayjs().utc().format();
+    
+    if (Array.isArray(body["data"])) {
+      const page = serializePagination(req).page;
```

Comment:
This shouldn't be here, once the data processing has been completed, you should end up with 

```
{
  items: [],
  pagination
}
```
from the controller. 

From there you can determine that if:
- the items property exists, and it's a type of array
and
- the pagination property exists

You can then make use of the pagination property to get pagination information